caikit_nlp/modules/text_generation/peft_prompt_tuning.py

```diff
@@ -170,48 +174,51 @@ def __del__(self):
     def run(
         self,
         text: str,
-        device: Optional[Union[str, int]] = _DETECT_DEVICE,
-        max_new_tokens=20,
-        min_new_tokens=0,
+        max_new_tokens: Optional[int] = 20,
+        min_new_tokens: Optional[int] = 0,
+        truncate_input_tokens: Optional[int] = 0,
+        decoding_method: Optional[str] = "GREEDY",
+        top_k: Optional[int] = 0,
+        top_p: Optional[float] = 0.0,
+        typical_p: Optional[float] = 0.0,
+        temperature: Optional[float] = 1.0,
+        repetition_penalty: Optional[float] = 0.0,
```

Comment:
I think the default repetition penalty should be `1.0`! 1.0 means no penalty, and is the default for both the login processor in transformers and TGIS

https://huggingface.co/docs/transformers/main_classes/text_generation#transformers.GenerationConfig.repetition_penalty

https://github.com/IBM/text-generation-inference/blob/main/router/src/lib.rs#L80